src/MigrationTools.Clients.AzureDevops.Rest/Processors/AzureDevOpsPipelineProcessor.cs

```diff
@@ -205,6 +207,11 @@ private async Task<IEnumerable<Mapping>> CreateBuildPipelinesAsync(IEnumerable<M
                     .FirstOrDefault(c => c.Id == sourceConnectedServiceId)?.Name == s.Name)?.Id;
                 definitionToBeMigrated.Repository.Properties.ConnectedServiceId = targetConnectedServiceId;
 
+                var sourceRepoId = definitionToBeMigrated.Repository.Id;
```

Comment:
I'm not sure it's safe to assume `definitionToBeMigrated.Repository` is never null.. I wonder if all repo types have a "repository" property or not. Should you just check for null?

Reply:
I'll have to check that but if the source ID is null the target ID will should be nulled too

Reply:
If source definition's Repository Property is null it will throw an exception trying to get it's ID.

Reply:
Is this mergable?

Reply:
> If source definition's Repository Property is null it will throw an exception trying to get it's ID.

@MrHinsh May I know why this PR was merged, despite of the above fact ?